### ISSUE TITLE
remove refs to people.apache.org/home.apache.org in build

### ISSUE THREAD

**Issue by rmuir** [apache/lucene#13647]
### Description

The server will be decomm'ed soon, we should fix these refs:

```
$ git grep people.apache.org
$ git grep home.apache.org
```

**Comment by mikemccand** [comment]
Thanks @rmuir and @ChrisHegarty.

I've downloaded all my content from `home.apache.org` (Lucene benchmark source corpora, line file docs, large vector file, etc.), so we won't lose any benchy stuff once the box goes poof.

I need to find a new home for the nightly benchmarks logs/charts, currently at https://home.apache.org/~mikemccand/lucenebench ... when each nightly benchy finishes it copies up the results (all charts with a new data point) using sftp via Python.  I'm leaning towards a [simple GitHub pages site](https://docs.github.com/en/pages) (thank you @msokolov for the idea), though it has a limit of 1 GB and the benchy reports are now ~1.7 GB: ~13 years of detailed nightly benchy reports adds up!  I can probably work around that.

For the larger stuff (corpora) I'll mull some more.  I have plenty of storage in my personal Google drive account, so I can just start there.

**Comment by mikemccand** [comment]
> I'm leaning towards a [simple GitHub pages site](https://docs.github.com/en/pages) (thank you @msokolov for the idea)

I enabled pages for the `luceneutil` repro and pushed a copy of the current nightly benchy reports: https://mikemccand.github.io/luceneutil/index.html.  Looks like it basically works, yay!

It should be simple to fix the nightly benchy script to publish updates via `git add/commit/push` instead of the current `sftp mikemccand@home.apache.org`.  I'll do that next...

**Comment by mikemccand** [comment]
A nice side effect of this is that the long running (13+ years now!) nightly reports will be backed up via git/GitHub and no longer single sourced on my home box, yay.  And if ever some exotic bug shows up in the publishing, we will have the full `git` history showing each nightly benchy update going forwards to help debug.

**Comment by msokolov** [comment]
Nice! glad it worked. 

FYI: I clicked on a few random links and found a 404 https://mikemccand.github.io/luceneutil/analyzers.html although this page does seem to exist on the current site

**Comment by mikemccand** [comment]
Phew, OK, I think nightly benchy is now successfully publishing automatically to https://mikemccand.github.io/lucenenightly (using GitHub pages).  Last night's run "just worked".

I had to create a new public repo (https://github.com/mikemccand/lucenenightly) just for publishing the many (~1.9 GB, ~5.9K pages) nightly artifacts over 13+ years.

> FYI: I clicked on a few random links and found a 404 https://mikemccand.github.io/luceneutil/analyzers.html although this page does seem to exist on the current site

Hmm I'll dig into this -- it's still an issue with `lucenenightly` repo too.  Thanks for catching it @msokolov.

**Comment by mikemccand** [comment]
> > FYI: I clicked on a few random links and found a 404 https://mikemccand.github.io/luceneutil/analyzers.html although this page does seem to exist on the current site
> 
> Hmm I'll dig into this -- it's still an issue with `lucenenig`

OK should be fixed now!  https://mikemccand.github.io/lucenenightly/analyzers.html

**Comment by javanna** [comment]
We also have references in the `external-datasets.gradle`  file:

      name = "enwiki.random.lines.txt"
      src = "https://home.apache.org/~mikemccand/${name}.zst"

      name = "enwiki-20070527-pages-articles.xml"
      src = "https://home.apache.org/~dsmiley/data/${name}.bz2"

      name = "geonames_20130921_randomOrder_allCountries.txt"
      src = "https://home.apache.org/~dsmiley/data/${name}.bz2"

      name = "top.100k.words.de.en.fr.uk.wikipedia.2009-11"
      src = "https://home.apache.org/~rmuir/wikipedia/${name}.tar.bz2"

cc @dsmiley @rmuir @mikemccand

**Comment by dsmiley** [comment]
I rescued those files a few weeks ago.  If someone wants to host them, I can share if home.apache.org is gone.
* enwiki-20070527-pages-articles.xml bz2 -- 2.69 GB
* geonames_20130921_randomOrder_allCountries.txt bz2 -- 311 MB

**Comment by mikemccand** [comment]
I've also put these two files on `https://githubsearch.mikemccandless.com`:

```
enwiki-20120502-lines-1k-fixed-utf8-with-random-label.txt.lzma  5.9G
wikimedium500.tasks  289K
enwiki-20110115-lines-1k-fixed-with-random-labels.bin  25G
```

I will try to find a better permanent ish home for these sources ...

**Comment by iamsanjay** [comment]
I was trying to set the [luceneutil](https://github.com/mikemccand/luceneutil), ran the script.
```
python3 src/python/setup.py -download
```
It failed on one url where it, i believe, downloads the embeddings `enwiki-20120502-lines-1k-100d.vec`
https://github.com/mikemccand/luceneutil/blob/e2fb399c25ba29b9712c6566e03ecf6d0803e009/src/python/setup.py#L35

```
BASE_URL = 'https://home.apache.org/~mikemccand'
BASE_URL2 = 'https://home.apache.org/~sokolov'
BASE_URL3 = 'https://githubsearch.mikemccandless.com'
DATA_FILES = [
  ('enwiki-20120502-lines-1k-fixed-utf8-with-random-label.txt.lzma', BASE_URL3),
  ('enwiki-20120502-lines-1k-100d.vec', BASE_URL2),
  #('enwiki-20120502-lines-1k-100d-8bit.vec', BASE_URL2),
  ('wikimedium500.tasks', BASE_URL3),
  ('glove.6B.zip', 'https://downloads.cs.stanford.edu/nlp/data/')
]
```

**Comment by dweiss** [comment]
@mikemccand would you be able to expose the files @dsmiley rescued on your server?

**Comment by dweiss** [comment]
I filed https://issues.apache.org/jira/browse/INFRA-26434 and asked if apache.org can be of any help here. Some of those files are too large to host on github (even in a separate repository, as lfs files). It would be good to have them stored somewhere though, so that benchmarks can be repeated.

**Comment by dsmiley** [comment]
[geonames_20130921_randomOrder_allCountries.txt.bz2](http://gofile.me/5MFBZ/edVjck97c) 297.2MB
If that works for you, I'll share the other.  If it doesn't I'll share in another way.

**Comment by dweiss** [comment]
Fetched, thanks, David. I'm talking to infra about the possibilities of storing those benchmark files somewhere on Apache services. I don't feel comfortable uploading it to github/gitlab/apache SVN, even if it's possible. Let's see what infra says. 

Some of those files could be generated on demand but then we'd lose the stable reference point if somebody wants to rerun them locally and compare the results.

**Comment by dweiss** [comment]
Thanks to INFRA-26434 Lucene now has an s3 bucket we can publish those data/test resources on. I'll try to collect these resources, upload them and make the necessary build changes so that everything works. I'll do this progressively and may ask you folks to give me access to those rescued files, when I need them.

**Comment by dweiss** [comment]
We now have an s3 bucket to place those benchmark/ reference files on. If you have any of these files - please let me know and perhaps make it available to me, somehow -
```
      name = "enwiki-20070527-pages-articles.xml"
      src = "https://home.apache.org/~dsmiley/data/${name}.bz2"

      name = "enwiki.random.lines.txt"
      src = "https://home.apache.org/~mikemccand/${name}.zst"

      name = "top.100k.words.de.en.fr.uk.wikipedia.2009-11"
      src = "https://home.apache.org/~rmuir/wikipedia/${name}.tar.bz2"
```

If there is anything else that any of you think should be on that s3 bucket, let me know as well, please. Thanks.

**Comment by rmuir** [comment]
@dweiss https://issues.apache.org/jira/secure/attachment/12429835/top.100k.words.de.en.fr.uk.wikipedia.2009-11.tar.bz2

**Comment by msokolov** [comment]
There are other vector data files - I think the key one that has become a reference point is Cohere 768d trained on wikipedia-derived docs, but I'm not sure where nightly benchmarks gets it from; maybe just a locally cached copy? I have a local copy I could share, but  can we attach 3G files here? And TBH it's possible nightlies is using an even larger one with the full 33M docs (I may have truncated the copy I have since I almost never test with more than 2M docs)

**Comment by benwtrent** [comment]
@msokolov the python script in Lucene util downloads from hugging face. If that is the data you are talking about?

`infer_token_vectors_cohere.py`

**Comment by dweiss** [comment]
> [...] but can we attach 3G files here? 

I think we can, if it makes sense to do so.

We're not supposed to abuse this service - for example by downloading 3gb data files on each build... But we should use it to upload/store stuff that serves as a reference for benchmarks or tests. If it doesn't matter that a test file changes over time, let's just download it from source (hugging face or whatever). But for certain things - benchmarks over time, tests that should reproduce - it would be nice to offer a source of the same identical file for everyone who is interested.

Test runner boxes (CI) can keep a local copy of those large files to prevent bandwidth abuse. Github workers should be configured not to download those huge files, etc.

**Comment by msokolov** [comment]
Yes, I was referring to files that can be generated with `infer_token_vectors_cohere.py`. Maybe we take the position that users should regenerate, but it is kind of slow and demands a lot of CPU, so my practice is to keep copying these files around.

**Comment by dweiss** [comment]
I can generate this file and make it available as a benchmark dataset. Or would you rather give me one of your own, for consistency with your previous results?

**Comment by dweiss** [comment]
There are two or three references in test files. There is one reference remaining in releaseWizard.py:
```
key_url = "https://home.apache.org/keys/committer/%s.asc" % id.strip()
```
this is the gpg configuration (keys were supposed to be placed on h.a.org). I'm not sure what to do with it.

**Comment by rmuir** [comment]
@dweiss we could fetch https://whimsy.apache.org/public/public_ldap_people.json and retrieve committer's GPG fingerprint that way?

**Comment by mikemccand** [comment]
Oooh we have an official S3 bucket to use now?  I had already uploaded the benchy corpus files to my own S3 bucket ... I think the URLs are in the setup.py (just renamed to `initial_setup.py`):

```
DATA_FILES = [
  # remote url, local name
  ('https://luceneutil-corpus-files.s3.ca-central-1.amazonaws.com/enwiki-20120502-lines-1k-fixed-utf8-with-random-label.txt.lzma', 'enwiki-20120502-l\
ines-1k-fixed-utf8-with-random-label.txt.lzma'),

  ('https://luceneutil-corpus-files.s3.ca-central-1.amazonaws.com/cohere-wikipedia-docs-768d.vec', 'cohere-wikipedia-docs-768d.vec'),

  ('https://luceneutil-corpus-files.s3.ca-central-1.amazonaws.com/cohere-wikipedia-queries-768d.vec', 'cohere-wikipedia-queries-768d.vec'),

  ('https://downloads.cs.stanford.edu/nlp/data/glove.6B.zip', 'glove.6B.zip'),
]
```

The Cohere docs file is quite large, like ~95 GB?

**Comment by mikemccand** [comment]
> [@mikemccand](https://github.com/mikemccand) would you be able to expose the files [@dsmiley](https://github.com/dsmiley) rescued on your server?

oh, hmm, not I haven't yet.  sorry, catching up.  eventually consistent Mike!

**Comment by dweiss** [comment]
Hmm... 100 gb may be stretching Apache Infra's patience... I don't even know if this bucket has a limit of some sort.

**Comment by msokolov** [comment]
I guess thius 94GB comes from 33M*768*4 bytes? Frankly I never test with indexes > ~2M docs, but maybe there is a call for the 33M-doc index in nightlies?

**Comment by mikemccand** [comment]
> I guess thius 94GB comes from `33M x 768 x 4` bytes? Frankly I never test with indexes > ~2M docs, but maybe there is a call for the 33M-doc index in nightlies?

Yeah ... nightly benchy builds a ~33M vectors index.  I think it's helpful to confirm we can index/search decent sized KNN indices ...

Maybe we just leave the 94 GB Cohere vectors source on my s3 bucket for now?  We can reassess if it becomes a problem ... `luceneutil` `initial_setup.py` should already be downloading from there...

**Comment by dweiss** [comment]
I've downloaded and moved all those data sets that were present in gradle build files (specifically, in external-datasets.gradle). If there is anything else I should place there, let me know. I agree those humongous files can probably stay on your AWS/benchmark machine since not many people on Earth have the processing power to deal with it... Certainly Robert's laptop isn't on the list of eligible hardware. :)
